cmd/nitro/nitro.go

```diff
@@ -405,19 +406,20 @@ func main() {
 		}
 
 		validatorNeedsKey := nodeConfig.Node.Validator.Enable && !strings.EqualFold(nodeConfig.Node.Validator.Strategy, "watchtower")
-		if nodeConfig.Node.BatchPoster.Enable || validatorNeedsKey {
+		if nodeConfig.Node.BatchPoster.Enable || validatorNeedsKey || setupNeedsKey {
+			daSigner, err = arbnode.GetSignerFromWallet(l1Wallet)
+			if err != nil {
+				fmt.Printf("%v\n", err.Error())
+				return
+			}
+
 			l1TransactionOpts, err = util.GetTransactOptsFromWallet(
 				l1Wallet,
 				new(big.Int).SetUint64(nodeConfig.L1.ChainID),
 			)
```

Comment:
At some point we really need to unify these code paths, but I don't think it's worth it for now